cpp/src/arrow/util/qpl_job_pool.h

```diff
@@ -0,0 +1,68 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <cstdint>
+#include <memory>
+#include <random>
+#include <utility>
+#include <vector>
+#include "arrow/status.h"
+
+#ifdef ARROW_WITH_QPL
+#include "qpl/qpl.h"
+#include "qpl/qpl.hpp"
+
+namespace arrow {
+namespace util {
+namespace internal {
+
+/// QplJobHWPool is resource pool to provide the job objects.
+/// Job object is used for storing context information during
+//  offloading compression job to HW Accelerator.
+class QplJobHWPool {
+ public:
+  QplJobHWPool();
+  ~QplJobHWPool();
+
+  static QplJobHWPool& instance();
```

Comment:
Function names here look more like the Java style. I'd suggest to follow the convention like `arrow/memory_pool.h` and add more comments about the public APIs. BTW, is it possible to refactor the static functions to be member functions? You may simply use a singleton to call member functions.  cc @pitrou

Reply:
@wgtmac I have addressed your comment including function name styles, comments, and static functions. Can you please take a look again? Thanks.